src/libclipper/test/serialization_test.cpp

```diff
@@ -14,15 +15,16 @@ namespace {
   const int NUM_STRING_INPUTS = 300;
 
   template<typename T>
-  void get_primitive_data_vectors(clipper::InputType type, std::vector<std::vector<T>> &data_vectors) {
+  std::vector<std::vector<T>> get_primitive_data_vectors() {
```

Comment:
This is clever, but I think it's overly complicated for test code. I'd rather have more verbose and simpler code in our unit tests. If a test starts failing, no one should be struggling to try to understand what the test code is doing.

If you make `j` a `uint8_t` then you're only ever casting from `uint8_t` to longer types which won't lose any information. Then you don't have to mess with pointers at all. Does that fix the problem?